### PR TITLE
fix: Use Object Manager for ModeConfigured

### LINKED ISSUE #38875: Object Manager not used for Magento\Csp\Model\Mode\Data\ModeConfigured
**File**
https://github.com/magento/magento2/blob/ef81f5a2f8271ea529606fcb56898eed286d46b5/app/code/Magento/Csp/Model/Mode/ConfigManager.php#L121

**Description**
The Object Manager is not being used when creating `Magento\Csp\Model\Mode\Data\ModeConfigured`. Therefore Plugins can't be used on this object.

### PR DESCRIPTION
**Pull request by pmzandbergen**
Fixing #38875 (AC-12299)

### Description (*)
Use Object Manager to create objects.

### Related Pull Requests
N/A

### Fixed Issues (if relevant)
1. Fixes magento/magento2#38875 (AC-12299)

### Manual testing scenarios (*)
N/A

### Questions or comments
Created this PR @glo24157 , feel free to modify or include with your changes. Also see comments at https://github.com/magento/magento2/commit/1b21cb81126d67b01bd01b4a39b7d1801dc68d0b

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by pmzandbergen** [comment]
@magento run all tests

File: app/code/Magento/Csp/Model/Mode/ConfigManager.php [hunk 1]
```diff
@@ -118,9 +126,9 @@ public function getConfigured(): ModeConfiguredInterface
             );
         }
 
-        return new ModeConfigured(
-            (bool) $reportOnly,
-            !empty($reportUri) ? $reportUri : null
-        );
+        return $this->modeConfiguredFactory->create([
+            'reportOnly' => $reportOnly,
```
Comment (Den4ik) [inline app/code/Magento/Csp/Model/Mode/ConfigManager.php]:
```suggestion
            'reportOnly' => (bool)$reportOnly,
```

**Review by Den4ik — CHANGES_REQUESTED** [review]
Hi @pmzandbergen 
Thanks for contribution. Please rolling back type casting for `$reportOnly` argument

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run Unit Tests

**Comment by Den4ik** [comment]
@magento run Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by pmzandbergen** [comment]
> Please rolling back type casting for $reportOnly argument

@Den4ik do you want me to roll back the type casting (and why)? The failing tests don't seem to be relevant to this change.

**Comment by Den4ik** [comment]
Failed tests seems are not related and looks strange that they are failed. Let's try one more time and compare results of failed tests

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run Unit Tests

File: app/code/Magento/Csp/Model/Mode/ConfigManager.php [hunk 2]
```diff
@@ -45,17 +50,20 @@ class ConfigManager implements ModeConfigManagerInterface
...
      */
     public function __construct(
         ScopeConfigInterface $config,
+        ModeConfiguredFactory $modeConfiguredFactory,
         Store $store,
         State $state,
         ?Http $request = null
     ) {
         $this->config = $config;
+        $this->modeConfiguredFactory = $modeConfiguredFactory;
```
Comment (Den4ik) [inline app/code/Magento/Csp/Model/Mode/ConfigManager.php]: Could you please update new constructor parameter based on [Backward compatibility documentation](https://developer.adobe.com/commerce/contributor/guides/code-contributions/backward-compatibility-policy/#adding-a-constructor-parameter)

**Review by Den4ik — CHANGES_REQUESTED** [review]
@pmzandbergen Could you please review my comment

**Comment by pmzandbergen** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run Unit Tests

**Comment by engcom-Dash** [comment]
@magento run all tests

File: app/code/Magento/Csp/Model/Mode/ConfigManager.php [hunk 3]
```diff
@@ -118,9 +127,9 @@ public function getConfigured(): ModeConfiguredInterface
             );
         }
 
-        return new ModeConfigured(
-            (bool) $reportOnly,
-            !empty($reportUri) ? $reportUri : null
-        );
+        return $this->modeConfiguredFactory->create([
+            'reportOnly' => (bool) $reportOnly,
+            'reportUri' => !empty($reportUri) ? (string) $reportUri : null,
```
Comment (Den4ik) [inline app/code/Magento/Csp/Model/Mode/ConfigManager.php]:
```suggestion
            'reportUri' => !empty($reportUri) ? (string)$reportUri : null
```

**Comment by Den4ik** [comment]
@magento run all tests

**Comment by Den4ik** [comment]
@magento run Unit Tests

**Comment by engcom-Dash** [comment]
@magento run all tests

**Review by Den4ik — APPROVED** [review]
✅ Approved.
Changes looks ok for me.
Failed tests each time are different and don't related to the changes.

**Comment by engcom-Dash** [comment]
@magento run all tests

**Comment by engcom-Dash** [comment]
Hello @pmzandbergen 

Thanks for the collaboration & contribution!

✔️ QA Passed
Preconditions:

Install fresh Magento 2.4-develop
Steps to reproduce

- Create a new plugin on method: Magento\Csp\Model\Mode\Data\ModeConfigured::isReportOnly()
- Set a breakpoint in the plugin's method or use the die keyword.
- Access the homepage URL or admin page URL.

Before: ✖️
homepage loads without triggering the plugin.
image

After: ✔️
Plugin is triggered.
image

Builds are failed. Hence, moving this PR to Extended Testing.

Thanks

**Comment by engcom-Dash** [comment]
@magento run all tests

**Comment by engcom-Dash** [comment]
I have a green build (all tests passed), so I'm moving it to Merge in Progress.

image

**Comment by engcom-Dash** [comment]
We have seen static failures in mainline PR after merging this PR into that. Hence for now moving this PR to Extended testing.

**Comment by engcom-Dash** [comment]
@magaento run all tests

**Comment by engcom-Dash** [comment]
@magento run all tests

**Comment by engcom-Dash** [comment]
@magento run Functional Tests CE, Functional Tests B2B

**Comment by engcom-Dash** [comment]
The Functional B2B and CE test failures seem flaky and some of these failures are known issues. They are neither part of PR now failing because of PR changes. Hence moving it to Merge in Progress.

**Functional Tests B2B**
Run 1: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/38886/6f2db52807a3c708a0be9960115b0689/Functional/allure-report-b2b/index.html#categories/3a3a8306986cbb01e5245b23e6e5b238/715420361f789b7/
image

Run 2: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/38886/37b223604492bd6d8496120f6e64de42/Functional/allure-report-b2b/index.html#categories/3a3a8306986cbb01e5245b23e6e5b238/1b73508e9f000b5f/
image

Known Issues:
MultiShippingWithCreationNewCustomerAndAddressesDuringCheckoutTest: [ACQE-7229](https://jira.corp.adobe.com/browse/ACQE-7229)
StoreFrontAddOutOfStockProductToShoppingCartTestCest: [ACQE-6818](https://jira.corp.adobe.com/browse/ACQE-6818)
StorefrontReuseCouponCodeAfterOrderCanceledTest: [ACQE-7230](https://jira.corp.adobe.com/browse/ACQE-7230)

**Functional Tests CE**
Run 1: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/38886/42b5eb56906513fe5eea7fe14dd341e3/Functional/allure-report-ce/index.html#categories/6d7e37752c3cc83cb3d2063131f608bb/93d477020fb48965/
image

Run 2: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/38886/974ba1a9f4df8f43176d877447c86fcf/Functional/allure-report-ce/index.html#categories/6d7e37752c3cc83cb3d2063131f608bb/7b4b2f91ca0a480a/
image

Known Issue: 
VerifyDefaultWYSIWYGToolbarOnProductTest: [ACQE-7227](https://jira.corp.adobe.com/browse/ACQE-7227)

**Pull request merged**